### DISCUSSION TITLE
CMake Error when 'cmake . -B build' : The CUDA compiler identification is unknown

### DISCUSSION THREAD

**Discussion by mm23mm** [NVlabs/instant-ngp#285] [Q&A]
Visual Studio 2019
CUDA v11.6
 When i use the command 'cmake . -B build', i got the following information:
![image](https://user-images.githubusercontent.com/92322655/156008970-19f3e51c-94eb-42ff-b7de-c2d29740c8f0.png)

And i have configured the environment variables
![image](https://user-images.githubusercontent.com/92322655/156009160-1bc375ff-ea12-471f-954e-dee90618a1a4.png)

I don't know how to solve this problem

**Comment by leventt** [comment]
You should try re-installing CUDA SDK if you haven't and try to build on Windows using x64 Native Developer Tools command line variant you can find in start menu instead of cmd if you aren't using that already. Installing CUDA and then Visual Studio in that order, will cause CUDA installer to not install some build extensions for Visual Studio for example, which may cause this... But these are me speculating without better logs.

Easiest when you get `No CMAKE_CUDA_COMPILER could be found` error like that is to run `nvcc -v` in that same command line session to make sure you have that binary in your path.
